spec/rspec/matchers/built_in/yield_spec.rb

```diff
@@ -335,6 +353,16 @@ def each_arg(*args, &block)
       }.to fail_with(/expected given block not to yield with arguments, but did/)
     end
 
+    it 'fails if the matchers match at yield time only' do
+      expect {
+        expect { |b|
+          val = []
+          _yield_with_args(val, &b)
+          val << 1
+        }.not_to yield_with_args(be_empty)
+      }.to fail_with(/expected given block not to yield with arguments, but did/)
```

Comment:
...and this seems like the spot where you could cover the negative failure message.